zerver/lib/markdown/__init__.py

```diff
@@ -2026,7 +2035,13 @@ def handleMatch(  # type: ignore[override] # https://github.com/python/mypy/issu
         el.set("data-stream-id", str(stream_id))
         stream_url = encode_stream(stream_id, stream_name)
         topic_url = hash_util_encode(topic_name)
-        link = f"/#narrow/stream/{stream_url}/topic/{topic_url}"
+        channel_topic_object = ChannelTopicInfo(stream_name, topic_name)
+        with_operand = self.get_with_operand(channel_topic_object)
+        if with_operand is not None:
+            link = f"/#narrow/stream/{stream_url}/topic/{topic_url}/with/{with_operand}"
+        else:
+            link = f"/#narrow/stream/{stream_url}/topic/{topic_url}"
```

Comment:
I posted in https://chat.zulip.org/#narrow/stream/378-api-design/topic/URLs.20for.20stream.20to.20channel.20rename/near/1953875 about this, but we might want to do #30385 as a quick prep commit for this, just to avoid needing to deal with `/stream/` and `/with/` in the same links.